docs/advanced_examples/KNearestNeighbors.ipynb

```diff
@@ -0,0 +1,296 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# K-Nearest Neighbors\n",
+    "\n",
+    "\n",
+    "K-Nearest Neighbors (KNN) is a non-parametric supervised learning algorithm, suitable for both classification and regression tasks.\n",
+    "\n",
+    "In classification, KNN aims to identify the nearest points by measuring their similarity, often through distance metrics. The new labels are then assigned through majority voting, considering the most frequent labels among the neighboring points.\n",
+    "\n",
+    "In Fully Homomorphic Encryption (FHE), classification with KNN poses significant computational challenges due to the distance calculations and the sorting algorithms, which is currently a non-stable algorithm.\n",
+    "\n",
+    "It is therefore recommended to use it on small datasets (up to dozens of examples) with strong quantization (n_bits <= 4).\n",
+    "\n",
+    "This tutorial focuses exclusively on the KNN classifier, comparing the performance of Concrete KNN with its scikit-learn counterpart on a small data-set."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Import libraries\n",
+    "\n",
+    "First, import the required packages, the classical KNN regressor and its Concrete ML counterpart."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import time\n",
+    "\n",
+    "import pandas as pd\n",
+    "from sklearn.datasets import make_classification\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "from tqdm.auto import tqdm\n",
+    "\n",
+    "from concrete.ml.sklearn import KNeighborsClassifier as ConcreteKNeighborsClassifier"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Data-set generation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X, y = make_classification(\n",
+    "    n_samples=13, n_features=20, n_informative=3, n_redundant=0, n_classes=2, n_clusters_per_class=1\n",
+    ")\n",
+    "# Split the data-set into a train and testing sets\n",
+    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Model instantiation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# The novel aspect introduced by Concret-ML models is the hyperparameter `n_bits`, which represents\n",
+    "# the precision for quantizing input data and intermediate values of the model.\n",
+    "# This quantization step is essential after the training phase, since FHE exclusively operates\n",
+    "# over integers.\n",
+    "concrete_knn = ConcreteKNeighborsClassifier(n_bits=3, n_neighbors=3)\n",
```

Comment:
```
and intermediate values of the model
```
I don't think this is true? The accumulator depends on the n_bits but the precision of intermediate values will differ based on the operations.

I think we can have a shorter here that just says input is quantized to n_bits.